authentication/views.py

```diff
@@ -125,3 +127,15 @@ def put(self, request):
                 return Response({'msg': 'Invalid verification token!'}, status=400)
         else:
             return Response({'msg': "User doesn't exist!"}, status=400)
+
+class UpdateProfileView(APIView):
+    permission_classes = [IsAuthenticated]
+    def put(self, request):
+        user = request.user
+        profile = user.profile
+        serializer = ProfileSerializer(instance=profile, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response({'msg': 'Profile updated successfully!'})
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Change the file to be ProfileView(APIView). Because u might want to get user profile so if that's the case, you will want to  add get method on the class.

For example:
```
class ProfileView(APIView):
    def get(self, request):
        # Implement the get logic here.
        
    def put(self, request):
        user = request.user
        profile = user.profile
        serializer = ProfileSerializer(instance=profile, data=request.data, partial=True)
        if serializer.is_valid():
            serializer.save()
            return Response({'msg': 'Profile updated successfully!'})
        else:
            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
This will make your code more efficient. Overall great implementation, keep it up!